packages/cli/src/commands/build/index.ts

```diff
@@ -807,3 +811,21 @@ function mergeFlags(
     return [];
   });
 }
+
+function isUsingSpeedInsights(pkg: PackageJson | null): boolean {
+  const dependencies = [
+    ...Object.keys(pkg?.dependencies ?? {}),
+    ...Object.keys(pkg?.devDependencies ?? {}),
+  ];
+
+  return dependencies.some(d => d === '@vercel/speed-insights');
+}
+
+function isUsingWebAnaytics(pkg: PackageJson | null): boolean {
+  const dependencies = [
+    ...Object.keys(pkg?.dependencies ?? {}),
+    ...Object.keys(pkg?.devDependencies ?? {}),
+  ];
+
+  return dependencies.some(d => d === '@vercel/analytics');
+}
```

Comment:
maybe we can abstract away checking if a package is installed
```suggestion
function isPackageInstalled(name: string; pkg: PackageJson | null) {
  const dependencies = [
    ...Object.keys(pkg?.dependencies ?? {}),
    ...Object.keys(pkg?.devDependencies ?? {}),
  ];

  return dependencies.some(d => d === name);
}

function isUsingSpeedInsights(pkg: PackageJson | null): boolean {
  return isPackageInstalled('@vercel/speed-insights', pkg);
}

function isUsingWebAnaytics(pkg: PackageJson | null): boolean {
   return isPackageInstalled('@vercel/analytics', pkg);
}
```